File: easy/0083_remove_duplicates_from_sorted_list/solution.cpp
#include "../../utils/cpp/nodes.hpp"

class Solution {
public:
  utils::ListNode *deleteDuplicates(utils::ListNode *head) {
    utils::ListNode *curr = head;

    while (curr) {
      if (curr->next && curr->val == curr->next->val)
        curr->next = curr->next->next;
      else
        curr = curr->next;
    }

    return head;
  }
};

File: easy/0083_remove_duplicates_from_sorted_list/solution.py
from utils.python.nodes import ListNode


class Solution:
    def deleteDuplicates(self, head: ListNode | None) -> ListNode | None:
        curr = head

        while curr:
            if curr.next and curr.val == curr.next.val:
                curr.next = curr.next.next
            else:
                curr = curr.next

        return head
